pkg/querier/querier.go

```diff
@@ -321,34 +321,43 @@ func (q *Querier) Tail(ctx context.Context, req *logproto.TailRequest) (*Tailer,
 
 // passed to tailer for (re)connecting to new or disconnected ingesters
 func (q *Querier) tailDisconnectedIngesters(ctx context.Context, req *logproto.TailRequest, connectedIngestersAddr []string) (map[string]logproto.Querier_TailClient, error) {
-	tailClients := make(map[string]logproto.Querier_TailClient)
-	for i := range connectedIngestersAddr {
-		tailClients[connectedIngestersAddr[i]] = nil
+	// Build a map to easily check if an ingester address is already connected
+	connectedIngestersAddrs := make(map[string]bool)
```

Comment:
Could we change the name of this or of the argument? They're so similar that it's a bit hard to read.